lib/graphql/type/schema.ex

```diff
@@ -38,14 +44,24 @@ defmodule GraphQL.Schema do
     else
       typemap = Map.put(typemap, type.name, type)
       thunk_fields = GraphQL.Execution.Executor.maybe_unwrap(type.fields)
-      typemap = Enum.reduce(thunk_fields, typemap, fn({_,fieldtype},map) ->
-        reduce_types(map, fieldtype.type)
+      typemap = Enum.reduce(thunk_fields, typemap, fn({_,fieldtype},typemap) ->
```

Comment:
not loving the idea of continually redefining `typemap`, perhaps this could be better expressed as a pipeline?